source/cl/test/UnitCL/source/cl_intel_unified_shared_memory/usm_enqueue.cpp

```diff
@@ -47,6 +47,11 @@ struct USMCommandQueueTest : public cl_intel_unified_shared_memory_Test {
 // Test for invalid API usage of clEnqueueMemFillINTEL()
 TEST_F(USMCommandQueueTest, MemFill_InvalidUsage) {
   for (auto ptr : allPointers()) {
+    if (ptr == host_shared_ptr) {
+      // Shared pointers without devices can't be memfill'd
+      continue;
+    }
```

Comment:
Since this test is about reporting the right errors for invalid uses of `MemFill`, maybe we should attempt it anyway and make sure it does not crash?

Reply:
Does the updated test and comment make sense? I've reworded it, but I'm not sure how understandable it is.

Reply:
I'm confused now, I thought after our other conversation that this was meant to be valid, but we have a bug and don't support it and that's why we're skipping testing for now. When I look at the spec, I get that same impression. The spec doesn't say the queue needs to be associated with the allocation's device (you're right, there isn't any such device), the spec says the allocation needs to be accessible by the device associated with the queue (which does exist).

Reply:
Reworded the comment. host_shared_pointer has no associated device.

Reply:
> device is an optional OpenCL device ID to associate with the allocation. If device is NULL then the allocation is not associated with any device. Allocations with no associated device are accessible by the host and OpenCL devices in the context that have cross-device access capabilities.

None of our devices support the cross-device access capability (yet).

Reply:
Now it's starting to make sense to me, thanks. Looks good, then.